File: Java-Core/src/main/java/prompto/expression/ParenthesisExpression.java
package prompto.expression;

import prompto.error.PromptoError;
import prompto.error.SyntaxError;
import prompto.runtime.Context;
import prompto.type.IType;
import prompto.utils.CodeWriter;
import prompto.value.IValue;

public class ParenthesisExpression implements IExpression {

	IExpression expression;
	
	public ParenthesisExpression(IExpression expression) {
		this.expression = expression;
	}

	@Override
	public String toString() {
		return "(" + expression.toString() + ")";
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		writer.append("(");
		expression.toDialect(writer);
		writer.append(")");
	}
	
	@Override
	public IType check(Context context) throws SyntaxError {
		return expression.check(context);
	}
	
	@Override
	public IValue interpret(Context context) throws PromptoError {
		return expression.interpret(context);
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/expression/ParenthesisExpression.js
function ParenthesisExpression(expression) {
	this.expression = expression;
	return this;
};

ParenthesisExpression.prototype.toString = function() {
	return "(" + this.expression.toString() + ")";
};

ParenthesisExpression.prototype.toDialect = function(writer) {
    writer.append("(");
    this.expression.toDialect(writer);
    writer.append(")");
};

ParenthesisExpression.prototype.check = function(context) {
	return this.expression.check(context);
};

ParenthesisExpression.prototype.interpret = function(context) {
	return this.expression.interpret(context);
};

exports.ParenthesisExpression = ParenthesisExpression;
